### PR TITLE
Fix PSO bound type

### PR DESCRIPTION
**Pull request by zoq**
Fix for issue reported in https://github.com/mlpack/ensmallen/issues/123.

### CODE REVIEW

File: include/ensmallen_bits/pso/pso.hpp [hunk 1]
```diff
@@ -159,16 +159,16 @@ class PSOType
   size_t& NumParticles() { return numParticles; }
 
   //! Retrieve value of lowerBound.
-  size_t LowerBound() const { return lowerBound; }
+  arma::vec LowerBound() const { return lowerBound; }
```
Comment (lozhnikov) [inline include/ensmallen_bits/pso/pso.hpp]: Shouldn't it be a const reference?
```c++
const arma::vec& LowerBound() const { return lowerBound; }
```

File: include/ensmallen_bits/pso/pso.hpp [hunk 2]
```diff
@@ -159,16 +159,16 @@ class PSOType
...
 
   //! Modify value of lowerBound.
-  size_t& LowerBound() { return lowerBound; }
+  arma::vec& LowerBound() { return lowerBound; }
 
   //! Retrieve value of upperBound.
-  size_t UpperBound() const { return upperBound; }
+  arma::vec UpperBound() const { return upperBound; }
```
Comment (lozhnikov) [inline include/ensmallen_bits/pso/pso.hpp]: See the comment above.

File: include/ensmallen_bits/pso/pso.hpp (hunk 1 above)
Comment (zoq) [reply include/ensmallen_bits/pso/pso.hpp]: Right, updated.

File: include/ensmallen_bits/pso/pso.hpp [hunk 3]
```diff
@@ -159,16 +159,16 @@ class PSOType
   size_t& NumParticles() { return numParticles; }
 
   //! Retrieve value of lowerBound.
-  size_t LowerBound() const { return lowerBound; }
+  const arma::vec LowerBound() const { return lowerBound; }
```
Comment (lozhnikov) [inline include/ensmallen_bits/pso/pso.hpp]: Did you forget to add a reference `&`?
```c++
const arma::vec&
```
I thought the copy-constructor of `arma::vec` isn't trivial.

File: include/ensmallen_bits/pso/pso.hpp [hunk 4]
```diff
@@ -159,16 +159,16 @@ class PSOType
...
 
   //! Modify value of lowerBound.
-  size_t& LowerBound() { return lowerBound; }
+  arma::vec& LowerBound() { return lowerBound; }
 
   //! Retrieve value of upperBound.
-  size_t UpperBound() const { return upperBound; }
+  const arma::vec UpperBound() const { return upperBound; }
```
Comment (lozhnikov) [inline include/ensmallen_bits/pso/pso.hpp]: See the comment above.

File: include/ensmallen_bits/pso/pso.hpp (hunk 3 above)
Comment (zoq) [reply include/ensmallen_bits/pso/pso.hpp]: You are absolutely right, thanks again for pointing this out.

**Comment by rcurtin** [comment]
@lozhnikov thanks for catching the const reference issue; I completely overlooked it. :)

**Pull request merged**